sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/EndToEndTests.java

```diff
@@ -690,12 +690,23 @@ public void testIllegalDecimal() {
         this.testNegativeQuery(query, "DECIMAL type must have scale <= precision");
     }
 
+    // Tested on MySQL
     @Test
     public void decimalParse() {
         String query = "SELECT CAST('0.5' AS DECIMAL)";
         this.testConstantOutput(query, new DBSPZSetLiteral.Contents(
                 new DBSPTupleExpression(
-                        new DBSPDecimalLiteral(new DBSPTypeDecimal(CalciteObject.EMPTY, DBSPTypeDecimal.MAX_PRECISION, DBSPTypeDecimal.MAX_SCALE, false), new BigDecimal("0.5")))));
+                        new DBSPDecimalLiteral(new DBSPTypeDecimal(CalciteObject.EMPTY, DBSPTypeDecimal.MAX_PRECISION, 0, false), new BigDecimal("1")))));
```

Comment:
I think we should add a warning to the documentation, since this behavior may be surprising. It is to me. And it's just the opposite from strings, where VARCHAR is unlimited precision.